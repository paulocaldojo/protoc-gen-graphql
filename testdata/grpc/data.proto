syntax = "proto3";

package protoc_gen_graphql.test.grpc;

option go_package = "github.com/paulocaldojo/protoc-gen-graphql/testdata/grpc";

message RequestB {
  string id = 1;
}

message ResponseB {
  string id = 1;
}
