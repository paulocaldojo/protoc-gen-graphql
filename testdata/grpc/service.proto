syntax = "proto3";

package protoc_gen_graphql.test.grpc;

option go_package = "github.com/paulocaldojo/protoc-gen-graphql/testdata/grpc";

import "graphql/options.proto";
import "grpc/data.proto";

// This is a comment attached to the Service service.
// This is its second line.
//
// And this is its second paragraph.
service Service {
  // This is a comment attached to the Method method.
  // This is its second line.
  //
  // And this is its second paragraph.
  rpc Method(Request) returns (Response) {
    option (graphql.method) = { operation: "query" };
  }

  rpc MethodB(RequestB) returns (ResponseB) {
    option (graphql.method) = { operation: "query" };
  }

  rpc DeprecatedMethod(Request) returns (Response) {
    option deprecated = true;
    option (graphql.method) = { operation: "query" };
  }
}

message Request {
  message Foo {
    message Bar {
      string bar = 1;
    }
  }

  string string_field = 1;
  Foo foo = 2;
  repeated Foo.Bar bar = 3;
  map<string, EnumType> map_field = 4;
  oneof oneof_field {
    string first = 5;
    Foo second = 6;
  }

  string deprecated_field = 7 [deprecated = true];
}

message Response {
  string string_field = 1;
}

enum EnumType {
  UNKNOWN = 0;
  FIRST = 1;
  SECOND = 2;
}
