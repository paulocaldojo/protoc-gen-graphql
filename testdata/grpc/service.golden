# DO NOT EDIT! Generated by protoc-gen-graphql.

"""
This is a comment attached to the Service service.
This is its second line.

And this is its second paragraph.
"""
type ProtocGenGraphqlTestGrpc_Service_Query {
  """
  This is a comment attached to the Method method.
  This is its second line.

  And this is its second paragraph.
  """
  method(input: ProtocGenGraphqlTestGrpc_RequestInput!): ProtocGenGraphqlTestGrpc_Response
  methodB(input: ProtocGenGraphqlTestGrpc_RequestBInput!): ProtocGenGraphqlTestGrpc_ResponseB
  deprecatedMethod(input: ProtocGenGraphqlTestGrpc_RequestInput!): ProtocGenGraphqlTestGrpc_Response @deprecated
}

type ProtocGenGraphqlTestGrpc_Request {
  stringField: String!
  foo: ProtocGenGraphqlTestGrpc_Request_Foo!
  bar: [ProtocGenGraphqlTestGrpc_Request_Foo_Bar!]!
  mapField: [ProtocGenGraphqlTestGrpc_Request_MapFieldEntry!]!
  oneofField: ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof
  deprecatedField: String! @deprecated
}

"""
`ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof` represents the `oneof_field` oneof in `protoc_gen_graphql.test.grpc.Request`.
"""
union ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof = ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_First | ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_Second

"""
`ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_First` represents the `first` oneof field in `protoc_gen_graphql.test.grpc.Request`.
"""
type ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_First {
  _typename: String
  first: String!
}

"""
`ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_Second` represents the `second` oneof field in `protoc_gen_graphql.test.grpc.Request`.
"""
type ProtocGenGraphqlTestGrpc_Request_OneofFieldOneof_Second {
  _typename: String
  second: ProtocGenGraphqlTestGrpc_Request_Foo!
}

input ProtocGenGraphqlTestGrpc_RequestInput {
  stringField: String
  foo: ProtocGenGraphqlTestGrpc_Request_FooInput
  bar: [ProtocGenGraphqlTestGrpc_Request_Foo_BarInput!]
  mapField: [ProtocGenGraphqlTestGrpc_Request_MapFieldEntryInput!]
  oneofField: ProtocGenGraphqlTestGrpc_Request_OneofFieldOneofInput
  deprecatedField: String
}

input ProtocGenGraphqlTestGrpc_Request_OneofFieldOneofInput {
  first: String
  second: ProtocGenGraphqlTestGrpc_Request_FooInput
}

type ProtocGenGraphqlTestGrpc_Request_Foo {
  _empty: Boolean
}

input ProtocGenGraphqlTestGrpc_Request_FooInput {
  _empty: Boolean
}

type ProtocGenGraphqlTestGrpc_Request_Foo_Bar {
  bar: String!
}

input ProtocGenGraphqlTestGrpc_Request_Foo_BarInput {
  bar: String
}

"""
`ProtocGenGraphqlTestGrpc_Request_MapFieldEntry` represents the `map_field` map in `protoc_gen_graphql.test.grpc.Request`.
"""
type ProtocGenGraphqlTestGrpc_Request_MapFieldEntry {
  key: String!
  value: ProtocGenGraphqlTestGrpc_EnumType!
}

"""
`ProtocGenGraphqlTestGrpc_Request_MapFieldEntryInput` represents the `map_field` map in `protoc_gen_graphql.test.grpc.Request`.
"""
input ProtocGenGraphqlTestGrpc_Request_MapFieldEntryInput {
  key: String
  value: ProtocGenGraphqlTestGrpc_EnumType
}

type ProtocGenGraphqlTestGrpc_Response {
  stringField: String!
}

enum ProtocGenGraphqlTestGrpc_EnumType {
  UNKNOWN
  FIRST
  SECOND
}
